upstream/helm.go

```diff
@@ -0,0 +1,148 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package upstream
+
+import (
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/blang/semver"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+
+	"helm.sh/helm/v3/pkg/cli"
+	"helm.sh/helm/v3/pkg/getter"
+	"helm.sh/helm/v3/pkg/repo"
+)
+
+// Helm upstream representation
+type Helm struct {
+	Base `mapstructure:",squash"`
+
+	// Helm repository URL, e.g. https://grafana.github.io/helm-charts
+	Repo string
+
+	// Helm chart name in this repository
+	Chart string
+
+	// Optional: semver constraints, e.g. < 2.0.0
+	// Will have no effect if the dependency does not follow Semver
+	Constraints string
+}
+
+// LatestVersion returns the latest non-draft, non-prerelease Helm Release
+// for the given repository (depending on the Constraints if set).
+func (upstream Helm) LatestVersion() (string, error) {
+	log.Debug("Using Helm flavour")
+	return latestChartVersion(upstream)
+}
+
+func latestChartVersion(upstream Helm) (string, error) {
+	// Sanity checking
+	if upstream.Repo == "" {
+		return "", errors.Errorf("Invalid helm upstream: missing repo argument")
+	}
+
+	if upstream.Chart == "" {
+		return "", errors.Errorf("Invalid helm upstream: missing chart argument")
+	}
+
+	if !strings.Contains(upstream.Repo, "//") {
```

Comment:
Do we wanna be more constraint here with:
```suggestion
	if !strings.HasPrefix(upstream.Repo, "https://") || !strings.HasPrefix(upstream.Repo, "http://") {
```